modules/tensor_mechanics/examples/piston/piston_params.i

```diff
@@ -0,0 +1,178 @@
+## This example is documented on YouTube at:
+## https://www.youtube.com/watch?v=L9plLYkAbvQ
+## 
+## Additional files (e.g. the CAD model, results)
+## can be downloaded freely from Zenodo at:
+## https://doi.org/10.5281/zenodo.3886965
+
+[Mesh]
+  displacements = 'disp_x disp_y disp_z' #Define displacements for deformed mesh
+  type = FileMesh #Read in mesh from file
+  file = piston_coarse.e
+[]
+
+# This is where mesh adaptivity magic happens:
+[./Adaptivity]
+  steps = 1
+  #initial_steps = 2
+  max_h_level = 3
+  cycles_per_step = 1
+  initial_marker = uniform
+  marker = errorFraction
+  [./Markers]
+    [./uniform]
+      type = UniformMarker
+      mark = refine
+    [../]
+    
+    [./errorFraction]
+      type = ErrorFractionMarker
+      coarsen = 0.5
+      indicator = gradientJump
+      refine = 0.5
+    [] 
+ [../]
+
+  [./Indicators]
+    [./gradientJump]
+      type = GradientJumpIndicator
+      variable = disp_y
+    []
+  []
+[../]
```

Comment:
I think we still have some issues with using adaptivity on mechanics problems. 

If you want to use adaptivity, the safest way is to use total small deformation that does not need stateful properties.

Reply:
I would like to use adaptivity in this example, can you suggest an adaptivity block?

Reply:
In the TensorMechanicsMasterAction, use `incremental = false` and `strain=SMALL`. Then it will not use any stateful properties.

Reply:
How's this https://github.com/idaholab/moose/pull/18543/commits/bad294ccccf393bf9d440f0003ebfa4b6a8262e8?

Reply:
could you run `hit format` on the input?

Reply:
@dschwen - see https://github.com/idaholab/moose/pull/18543/commits/72d4a16b0e5f22bb204e73df38d448bab22588d9